Add CHANGELOG entries for 1.16.0 release (#3549)
bottlerocket-os/bottlerocket: CHANGELOG.md

# v1.16.0 (2023-10-25)

## OS Changes

> * Set default netlink timeout to infinity to prevent interfaces from entering a failed state ([#3520])

Reviewer: nit: a bit more concise:
```suggestion
* Adjust netlink timeout to prevent interfaces from entering a failed state ([#3520])
```

> * Update third party packages ([#3535])

Reviewer: ```suggestion
* Update third-party packages ([#3535])
```

> * Add XFS CLI utilities for managing XFS-formated storage ([#3444])

Reviewer: ```suggestion
* Add XFS CLI utilities for managing XFS-formatted storage ([#3444])
```
